crates/client/settlement_client/src/l1_messaging.rs

```diff
@@ -54,9 +33,11 @@ pub async fn sync(
             return Err(e.into());
         }
     };
-    let event_filter = client.l1_core_contract.event_filter::<StarknetCoreContract::LogMessageToL2>();
 
-    let mut event_stream = event_filter
+    let contract_instance = client.get_core_contract_instance();
+    let event_filter = contract_instance.event_filter::<LogMessageToL2>();
```

Comment:
So I tried the other method but the thing is for CoreContractClient type we need to implement this otherwise the code won't compile saying that there is no method named `event_filter` for settlement_client.